### DISCUSSION TITLE
How should I modify the ekf algorithm so that ekf can fuse barometer and tof data at the same time?

### DISCUSSION THREAD

**Discussion by czyv587** [bitcraze/discussions#918] [Q&A]
If the drone only uses the tof sensor, it will jump up or down when passing a table or other obstacles. If the drone only uses the barometer, it will encounter the problem of inaccurate height and speed estimation when it is close to the ground. I want to combine the barometer and the tof datas in ekf at the same time to solve the problem of sudden height change and inaccurate height and speed estimation. How do I need to modify it in ekf?

**Comment by krichardsson** [comment]
You might be interested in #895
